### PR TITLE
Add unravel_index

### PR DESCRIPTION
**Pull request by diorcety**
Partial support of unravel_index

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
The ``ndarray`` type changes with #949 . i'd rather wait until it's merged to review this one :-)

File: pythran/pythonic/include/numpy/unravel_index.hpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#include "pythonic/include/types/str.hpp"
+
+PYTHONIC_NS_BEGIN
+
+namespace numpy
+{
+  template <class E, class IntTy, size_t N>
+  typename std::enable_if<std::is_scalar<E>::value,
+                          types::array<IntTy, N>>::type
+  unravel_index(E const &expr, types::array<IntTy, N> const &shape,
```
Comment (serge-sans-paille) [inline pythran/pythonic/include/numpy/unravel_index.hpp]: In order to match the new pshape scheme, ``types::array<IntTy, N>`` should be a ``Ss`` template, and N would be ``std::tuple_size<Ss>::value``

File: pythran/pythonic/numpy/unravel_index.hpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+{
+  namespace
+  {
+    template <class E, class ShapeIt, class RetIt>
+    void _unravel_index(E expr, ShapeIt shape_it, ShapeIt end_it, RetIt ret_it)
+    {
+      while (shape_it != end_it) {
+        auto &v = *shape_it;
+        *ret_it = expr % v;
+        expr /= v;
```
Comment (serge-sans-paille) [inline pythran/pythonic/numpy/unravel_index.hpp]: Did you benchmark your implementation vs. numpy version? this modulo + divide looks very costly to me.

File: pythran/tests/test_numpy_func3.py [hunk 3]
```diff
@@ -312,6 +312,12 @@ def test_unwrap1(self):
     def test_unwrap2(self):
         self.run_test("def np_unwrap2(x): from numpy import unwrap, pi ; x[:3] -= 2*pi; return unwrap(x, 4)", numpy.arange(6, dtype=float), np_unwrap2=[NDArray[float,:]])
 
+    def test_unravel_index_0(self):
+        self.run_test("def np_unravel_index0(x, y): from numpy import unravel_index; return unravel_index(x, y)", 1621, (6, 7, 8, 9), np_unravel_index0=[int, Tuple[int, int, int, int]])
+
+    def test_unravel_index_1(self):
+        self.run_test("def np_unravel_index1(x, y): from numpy import unravel_index; return unravel_index(x, y, 'F')", 1621, (6, 7, 8, 9), np_unravel_index1=[int, Tuple[int, int, int, int]])
+
```
Comment (serge-sans-paille) [inline pythran/tests/test_numpy_func3.py]: missing a 2 or 3 D test case.

Comment (diorcety) [reply pythran/tests/test_numpy_func3.py]: Sure. I was based on "who can do more can do less"

**Comment by serge-sans-paille** [comment]
Obsoleted by #995

**Pull request closed without merging**